plugin/federation/federation.go

```diff
@@ -195,16 +201,29 @@ func (f *federation) GenerateCode(data *codegen.Data) error {
 		}
 		for _, e := range f.Entities {
 			obj := data.Objects.ByName(e.Def.Name)
+			e.Type = types.TypeString(obj.Type, func(i *types.Package) string {
+				// TODO: somehow this must recognize case of `code.ImportPathForDir(s.destDir) == path`
+				f.imports = append(f.imports, i.Path())
+				if pkg := data.Config.Packages.NameForPackage(i.Path()); pkg != data.Config.Federation.Package {
+					return pkg
+				}
+				return ""
+			})
+			if e.Type == "" {
+				panic("unknown type " + obj.Type.String())
+			}
 
-			// fill in types for key fields
-			//
-			for _, keyField := range e.KeyFields {
-				if len(keyField.Field) == 0 {
-					fmt.Println("skipping key field " + keyField.Definition.Name + " in " + e.Def.Name)
-					continue
+			for _, r := range e.Resolvers {
+				// fill in types for key fields
+				//
+				for _, keyField := range r.KeyFields {
+					if len(keyField.Field) == 0 {
+						fmt.Println("skipping field " + keyField.Definition.Name + " in " + r.ResolverName + " in " + e.Def.Name)
```

Comment:
Is this output needed?

Reply:
i'd think so.  I think it would be useful to log why we are skipping a `@key`.  tho I would treat this more as an error, but leaving as is for now is good idea.

Reply:
@carldunham let's add the word `key` in the message again.  It makes it more clear what it is we are skipping.  Below in line 217 we include `requires` to disambiguate.